lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -1585,9 +1585,44 @@ SPIRVWord LLVMToSPIRV::transFunctionControlMask(Function *F) {
   return FCM;
 }
 
+void LLVMToSPIRV::transGlobalAnnotation(GlobalVariable *V) {
+  SPIRVDBG(dbgs() << "[transGlobalAnnotation] " << *V << '\n');
+
+  // @llvm.global.annotations is an array that contains structs with 4 fields.
+  // Get the array of structs with metadata
+  ConstantArray *CA = cast<ConstantArray>(V->getOperand(0));
+  for (Value *Op : CA->operands()) {
+    ConstantStruct *CS = cast<ConstantStruct>(Op);
+    // The first field of the struct contains a pointer to annotated variable
+    Value *AnnotatedVar = CS->getOperand(0)->getOperand(0);
```

Comment:
I guess you assume `CS->getOperand(0)` is a bitcast instruction. Will it work if we annotate a char ?

Reply:
Thank you for noticing, this code will not work properly with char variables :)

But I don't understand what is the advantage of using ->stripPointerCasts() instead just ->getOperand(0). Does it improve readability?